### PR TITLE
README.md minor corrections, partial major edition

### PR DESCRIPTION
**Pull request by grahamperrin**
Essentially: Mozilla bug 1568155 became a duplicate of 1417183. 

Whilst making that minor update, I took the opportunity to make: 

1. a handful of corrections to spelling
2. various changes to grammar
3. suggested improvements to overall readability, including conciseness. 

If my edits for areas (2) and (3) are excessive: apologies! Please cherry-pick whatever suits you. 

I should describe this as a _partial_ major edition, because I have not (yet) paid attention to the page as a whole. (It's now time for me to shift focus to my day job …) I'm happy to discuss; to _kick the ball around_, if you like.

### CODE REVIEW

**Comment by piroor** [comment]
Thank you for feedback! Your draft tells me more natural wordings.

I have a question about a wording about "add-on" and "extension": which is better word to call this product?

I thought that "add-on" looks better for a long time since Firefox 2.0: "Add-on" looks the name of a general category of software including both "extension" and "theme", and I thought that Mozilla want to use the term "add-on" instead of more detailed term "extension" in many situations, for example "Add-ons Manager", "Mozilla Add-ons" and others. However, today I cannot see no strict wording rule about these two names. For example, there is a menu command "Add-ons and Theme" in Firefox's application menu, it sounds that "add-on" is a category name with a level same to the one of "theme". However, the opened page "about:addons" contains categories "Extensions" and "Themes", this sounds that "add-on" is the name of an upper level category...

Today the website addons.mozilla.org uses the term "extension" widely and publicly to identify this kind software, so replacements of the term "add-on" to "extension" in this document looks reasonable. Did you think so?

**Comment by NiklasGollenstede** [comment]
I think Mozilla just used to call them Add-Ons, and (all?) other browsers/Chrome called them extensions.
Then Mozilla pushed WebExtensions, based on Chromium's extension architecture, and at the same time also started calling them extensions.
So, now "(browser) extension" seems the correct term for most situations.

---

Not a native English speaker myself, but FWIW I had a look at the edits a while ago, and they do read clearer / more fluently, and I think they also keep what you want to say.

**Comment by grahamperrin** [comment]
Hi 

In the context of 'add-ons': Mozilla's use of the word 'extensions' predates WebExtensions APIs. Consider the address – **about:addons** – and the _representation of_ types therein. 

----

Via the **Add-ons Support** link at the foot of about:addons in Thunderbird, from [Thunderbird add-ons - frequently asked questions](https://mzl.la/2cM7feG): 

> … Add-ons are extensions and other additions
> 
> … extensions … functionality …
> … themes … appearance …

----

Back to Firefox. 

Blurring the lines a little, [Add-on List-o-matic 9000](https://addons.mozilla.org/addon/add-on-list-o-matic-9000/) lists _extensions_ (not all types of add-on) including installed _search engines_. 

At https://addons.mozilla.org/en-GB/firefox/extensions/ we have: 

* extensions
* themes
* dictionaries
* language packs

image

– consistent with the essence of about:addons

image

– there's a link to **about:preferences** then from there, the link back to about:addons is described as **Extensions & Themes**: 

image

– and if I recall correctly, this simple distinction – **two types of add-on** – reflects a wish to simplify how things are perceived. 

### Summary

Extensions, dictionaries and language packs extend functionality. 

Themes should alter appearance without affecting functionality. 

Plug-ins are exceptional. https://developer.mozilla.org/en-US/docs/Plugins/Roadmap and so on.

Bottom line: 

* it's probably most useful to think of Tree Style Tab as an _extension_.

**Comment by grahamperrin** [comment]
https://github.com/piroor/treestyletab/pull/2877#issuecomment-841869677

> … I think they also keep what you want to say.

That's what most concerned me, given the scope of my edition. 

(It began as a single change to a bug number, I got carried away!)

I'll mark this PR as ready for review. 

A subsequent PR might attend to the page _as a whole_ … in my experience, this type of thing is easiest a few weeks later.

**Comment by piroor** [comment]
Sorry for this delay, my concerns disappeared and I've merged this proposal. Thanks a lot!

**Comment by grahamperrin** [comment]
Thanks! 

> Sorry for this delay, 

Not a problem. It's nice to take one's time.
